src/Components/RoleToolbar.js

```diff
@@ -0,0 +1,227 @@
+import {
+  Toolbar,
+  ToolbarContent,
+  ToolbarItem,
+  Dropdown,
+  DropdownItem,
+  DropdownToggle,
+  InputGroup,
+  TextInput,
+  ChipGroup,
+  Chip,
+  Select,
+  SelectOption,
+  DropdownToggleCheckbox,
+} from '@patternfly/react-core';
+import FilterIcon from '@patternfly/react-icons/dist/js/icons/filter-icon';
+import { capitalize } from '@patternfly/react-core/dist/esm/helpers/util';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+const RoleToolbar = ({
+  selectedRoles,
+  setSelectedRoles,
+  isChecked,
+  appSelections,
+  setAppSelections,
+  columns,
+  rows,
+  filteredRows,
+  pagedRows,
+  anySelected,
+  clearFiltersButton,
+  perPage,
+  nameFilter,
+  setNameFilter,
+  AccessRequestsPagination,
+  applications,
+}) => {
+    const [isBulkSelectOpen, setIsBulkSelectOpen] = React.useState(false);
+    // Filtering
+    const [isDropdownOpen, setIsDropdownOpen] = React.useState(false);
+    const [filterColumn, setFilterColumn] = React.useState(columns[0]);
+    const [isSelectOpen, setIsSelectOpen] = React.useState(false);
+    const selectLabelId = 'filter-application';
+    const selectPlaceholder = 'Filter by application';
```

Comment:
These values never change si its generaly better to declare them outside of the component. Right now the variable is re-declared on each render.